packages/@tailwindcss-postcss/src/index.ts

```diff
@@ -103,220 +103,239 @@ function tailwindcss(opts: PluginOptions = {}): AcceptedPlugin {
+                DEBUG && I.start('Lightning CSS')
+                let ast = optimizeCss(css, {
+                  minify: typeof optimize === 'object' ? optimize.minify : true,
+                })
+                DEBUG && I.end('Lightning CSS')
+
+                DEBUG && I.start('CSS -> PostCSS AST')
+                context.optimizedPostCssAst = postcss.parse(ast, result.opts)
+                DEBUG && I.end('CSS -> PostCSS AST')
+
+                DEBUG && I.end('Optimization')
+              } else {
+                // Convert our AST to a PostCSS AST
+                DEBUG && I.start('Transform Tailwind CSS AST into PostCSS AST')
+                context.cachedPostCssAst = cssAstToPostCssAst(tailwindCssAst, root.source)
+                DEBUG && I.end('Transform Tailwind CSS AST into PostCSS AST')
+              }
             }
-          }
 
-          context.tailwindCssAst = tailwindCssAst
+            context.tailwindCssAst = tailwindCssAst
+
+            DEBUG && I.start('Update PostCSS AST')
+            root.removeAll()
+            root.append(
+              optimize
+                ? context.optimizedPostCssAst.clone().nodes
+                : context.cachedPostCssAst.clone().nodes,
+            )
+
+            // Trick PostCSS into thinking the indent is 2 spaces, so it uses that
+            // as the default instead of 4.
+            root.raws.indent = '  '
+            DEBUG && I.end('Update PostCSS AST')
 
-          DEBUG && I.start('Update PostCSS AST')
-          root.removeAll()
-          root.append(
-            optimize
-              ? context.optimizedPostCssAst.clone().nodes
-              : context.cachedPostCssAst.clone().nodes,
-          )
+            DEBUG && I.end(`[@tailwindcss/postcss] ${relative(base, inputFile)}`)
+          } catch (error) {
+            // An error requires a full rebuild to fix
+            context.compiler = null
 
-          // Trick PostCSS into thinking the indent is 2 spaces, so it uses that
-          // as the default instead of 4.
-          root.raws.indent = '  '
-          DEBUG && I.end('Update PostCSS AST')
+            // Ensure all dependencies we have collected thus far are included so that the rebuild
+            // is correctly triggered
+            for (let file of context.fullRebuildPaths) {
+              result.messages.push({
+                type: 'dependency',
+                plugin: '@tailwindcss/postcss',
+                file: path.resolve(file),
+                parent: result.opts.from,
+              })
+            }
 
-          DEBUG && I.end(`[@tailwindcss/postcss] ${relative(base, inputFile)}`)
+            // We found that throwing the error will cause PostCSS to no longer watch for changes
+            // in some situations so we instead log the error and continue with an empty stylesheet.
+            console.error(error)
+            root.removeAll()
+          }
```

Comment:
I'd love if we could re-organize the code somehow so we're not wrapping 200 lines in a try/catch but this is probably fine for now.

ps you should be able to move the createCompiler() definition out of it